st2common/st2common/runners/base.py

```diff
@@ -172,26 +171,10 @@ def cancel(self):
         )
 
     def post_run(self, status, result):
-        # Handle callback (if specified)
-        callback = self.callback or {}
-
-        if callback and not (set(['url', 'source']) - set(callback.keys())):
-            callback_url = callback['url']
-            callback_module_name = callback['source']
-
-            try:
-                callback_module = register_callback_module(callback_module_name)
-            except:
-                LOG.exception('Failed importing callback module: %s', callback_module_name)
-
+        if self.callback and isinstance(self.callback, dict) and 'source' in self.callback:
+            callback_module = register_callback_module(self.callback['source'])
             callback_handler = callback_module.get_instance()
-
-            callback_handler.callback(
-                callback_url,
-                self.context,
-                status,
-                result
-            )
+            callback_handler.callback(self.liveaction)
```

Comment:
I also prefer older approach here - better decoupling. I'm not a fan of passing whole live action since it contains tons of attributes which could also change in the future.

Reply:
The older approach assumes that the only way to do callback is via a webhook.  The new approach lets the handler plugin decide how to do the callback either via a queue, webhook, etc.